src/System.Collections/src/System/Collections/Generic/HashSet.cs

```diff
@@ -1731,14 +1955,30 @@ private static bool AreEqualityComparersEqual(HashSet<T> set1, HashSet<T> set2)
         /// Workaround Comparers that throw ArgumentNullException for GetHashCode(null).
         /// </summary>
         /// <param name="item"></param>
+        /// <param name="comparer"></param>
         /// <returns>hash code</returns>
-        private int InternalGetHashCode(T item)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static int InternalGetHashCode(T item, IEqualityComparer<T>? comparer)
         {
             if (item == null)
             {
                 return 0;
             }
-            return _comparer.GetHashCode(item) & Lower31BitMask;
+
+            int hashCode = comparer?.GetHashCode(item) ?? item.GetHashCode();
+            return hashCode & Lower31BitMask;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Does this need really need to be marked with aggressive inlining?

Reply:
May not necessary for this one, but just keep it the same with other Internal* methods.